doc/ecs/create_own_content.md

```diff
@@ -0,0 +1,82 @@
+## Entitäten erstellen
+
+Anders als im klassischen OOP werden Entitäten nicht durch das Ableiten der `Entity` Klasse erzeugt und definiert, sondern durch das Hinzufügen von Komponente.
+
+Um eine eigene Entität zu erstellen, muss man:
+
+1. Eine neue Instanz vom Typ `Entity` anlegen
+2. Die gewünschten Komponenten erstellen
+3. Die Komponenten mit `Entity#addComponent` hinzufügen
+
+Beispiel: 
+```java
+Entity monster = new Entity();
+PositionComponent pc= new PositionComponent(monster);
+AIComponent ai = new AIComponent(monster);
+AnimationComponent ac = new AnimationComponent(monster);
+VelocityComponent vc = new AnimationComponent(monster);
+
+monster.addComponent(PositionComponent.name,pc);
+monster.addComponent(AIComponent.name,ai);
+monster.addComponent(AnimationComponent.name,ac);
+monster.addComponent(VelocityComponent.name,vc);
+```
+
+*Anmerkung: Die Komponenten wurden im Beispiel alle mit den jeweiligen Default-Werten initialisiert. Die Verwendung der anderen Konstruktoren, mit eigenen Werten, geht natürlich auch.* 
+
+## Component erstellen
+Um eigene Components zu implementieren, muss die eigene Klasse von der abstrakten Klasse  `Component` abgeleitet werden.
+
+Components werden im package `ecs.components` abgelegt und sollen den Namensschema `$WHAT_IS_THIS_COMPONENT$Component`folgen.
+Jedes Component braucht eine statische Variable `String name`, um den Component-Typen zu identifizieren. Dieser Name wird als Key-Value des HashSets für die, in den Entitäten gespeicherten, Components verwendet. 
+
+Jede Component-Instanz gehört zu genau einer Entitäs-Instanz. Eine Entitäts-Instanz kann einen Component-Typen nur einmal speichern.
+
+Um Components für die DSL verfügbar zu machen, siehe TBD
+
+## System erstellen
+Um eigenen Systeme zu implementieren, muss die eigene Klasse von `ECS_System` abgeleitet werden.
+
+Die Funktionalität des Systems wird in der `update`-Methode implementiert. Diese wird einmal pro Frame aufgerufen.
+
+In der `update`-Methode wird dann über das HashSet `ECS.entities`iteriert.
```

Comment:
Echt jetzt? Das muss jedes System nochmal selber machen? 

Die Iteration gehört für mich in die Basisklasse, und dort dann ein Aufruf einer abstrakten Methoden, die jedes neue System dann implementieren muss.

Reply:
Ich hab das mal nach #386 geschoben. Ich weiß aber nicht wie gut sich das mit der Streaming #270 kombinieren lässt. Muss man sich anschauen.